주문 통합테스트 구현

axhtl opened the pull request:
통합 테스트를 구현하였습니다.

OrderSeller
- [x] 판매자 주문 리스트 조회
- [x] 판매자 주문 승인
- [x] 판매자 주문 취소

OrderUser
- [x] 단일 주문 조회
- [x] 전체 주문 조회 offset 방식
- [x] 전체 주문 조회 slice 방식
- [x] 주문 생성
- [x] 주문상세 추가
- [x] 구매자 구매 확정
- [x] 구매자 주문 취소
- [x] 주문 삭제

axhtl committed:
test: 단일 주문 조회 통합테스트 생성

axhtl committed:
refactor: Optimize Imports & Reformat Code

axhtl committed:
refactor: 주석 삭제

axhtl committed:
refactor: DirtiesContext 어노테이션 추가

axhtl committed:
refactor: CreateOrderObject 파일 위치 이동

axhtl committed:
refactor: 단일 주문 통합테스트 오류 수정

axhtl committed:
test: OrderUser, OrderSeller 통합테스트 구현

axhtl committed:
refactor: 필요없는 메서드 삭제, Optimize Imports & Reformat Code

upswp commented on src/test/kotlin/org/store/clothstar/order/OrderSellerIntegrationTest.kt:
```diff
@@ -0,0 +1,164 @@
+        )
+
+        //then
+        actions.andExpect(status().isOk())
+            .andExpect(jsonPath("$.message").value("주문이 정상적으로 취소되었습니다."))
+
+        // 데이터베이스에서 주문 상태 조회하여 검증
+        val savedOrder: Order? = orderRepository.findByIdOrNull(order.orderId)
+        assertNotNull(savedOrder)
+        assertEquals(Status.CANCELED, savedOrder.status)
```
then 부분에는 기대하는 결과값에 대해서만 작성을 해두면 좋겠습니다!

upswp commented on src/test/kotlin/org/store/clothstar/order/OrderSellerIntegrationTest.kt:
```diff
@@ -0,0 +1,164 @@
+
+        // 데이터베이스에서 주문 상태 조회하여 검증
+        val savedOrder: Order? = orderRepository.findByIdOrNull(order.orderId)
+        assertNotNull(savedOrder)
```
null 체크를 하는 부분보다 객체에서 확인하고싶은 부분을 보시는게 좋습니다!

upswp commented on src/test/kotlin/org/store/clothstar/order/OrderSellerIntegrationTest.kt:
```diff
@@ -0,0 +1,164 @@
+        )
+
+        //then
+        actions.andExpect(status().isOk())
+            .andExpect(jsonPath("$.message").value("주문이 정상적으로 출고처리되었습니다."))
```
기대하는 부분은 아래로 내려주세요!

upswp commented on src/test/kotlin/org/store/clothstar/order/OrderSellerIntegrationTest.kt:
```diff
@@ -0,0 +1,164 @@
+        // 데이터베이스에서 주문 상태 조회하여 검증
+        val savedOrder: Order? = orderRepository.findByIdOrNull(order.orderId)
+        assertNotNull(savedOrder)
+        assertEquals(Status.PROCESSING, savedOrder.status)
```
각 단계별로 확실하게 확인하고 싶은 부분을 작성하신 부분은 좋습니다!

upswp commented on src/test/kotlin/org/store/clothstar/order/OrderUserIntegrationTest.kt:
```diff
@@ -0,0 +1,351 @@
+                .contentType(MediaType.APPLICATION_JSON)
+        )
+
+        //then
+        actions
+            .andExpect(status().isOk)
+            .andExpect(jsonPath("$.orderId").value(orderId))
```
조회의 특성을 살려 어떤 값을 조회했는지 예상한 값이 조회되었는지도 같이 작성되면 좋겠습니다.

upswp commented on src/test/kotlin/org/store/clothstar/order/OrderUserIntegrationTest.kt:
```diff
@@ -0,0 +1,351 @@
+        //then
+        actions.andExpect(status().isOk())
+            .andExpect(jsonPath("$.content.length()").value(10))
+    }
```
offset slice 부분이 나타나도록 코드를 다시 작성해보세요!

upswp commented on src/test/kotlin/org/store/clothstar/order/OrderUserIntegrationTest.kt:
```diff
@@ -0,0 +1,351 @@
+        // 데이터베이스에서 주문 조회하여 삭제 여부 검증
+        val savedOrder: Order? = orderRepository.findByIdOrNull(order.orderId)
+        assertNotNull(savedOrder)
+        assertNotNull(savedOrder.deletedAt)
```
시간값을 비교하는 부분이 추가되면 좋겠습니다!

upswp reviewed (changes requested):
고생하셨습니다! 수정한번 부탁드립니다!

axhtl committed:
refactor: 통합테스트 리팩토링

- then 부분에는 기대하는 결과값에 대해서만 작성
- offset과 slice의 특징이 나타나도록 테스트 검증
(findAllByOrderByOrderIdDesc() 메서드 추가)
- 주문 삭제에서 시간값을 비교하는 부분 추가
